SETUP/locale/compile.sh

```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 
-for po_file in `find . -name '*.po'`; do
-    mo_file=`echo $po_file | sed 's/.po/.mo/'`
-    echo Compiling $po_file to $mo_file
-    msgfmt $po_file -o $mo_file
-done
+while IFS= read -rd '' po_file; do
```

Comment:
Can we not with the IFS and really obscure bashisms here please? For loops are simple to read and understand.

Reply:
Yeah, for loops are simple to read, but they're also unsafe, which is kinda the problem that [SC2044](https://www.shellcheck.net/wiki/SC2044) is warning us about.

They break if you have arg separators or any metacharacters in any of your paths. Also `for f in $(find ...)` breaks if `find` returns a lot of output and exceeds your OS's command line arg limits, which is completely academic in our case, given `ARG_MAX` is typically >1MB these days, but it can be a problem. 

See also the long discussion on https://unix.stackexchange.com/a/9499

Reply:
For some of these cases we can get away with `find ... -exec sh -c 'shell commands'`: the overhead of starting a subshell is negligible, and we don't need to carry any variable/functions into the subshell. I'll go back and see the places I can do that simplification.

Reply:
Yes, but those are edgecases. The input to all of these scripts is largely well know. I'd rather kill SC2044 and keep the for loops because they are so much more readable. All of the fixes just make it so much more difficult to read and maintain.

Reply:
OK, done. The latest patchset goes back to suppressing the warning and using `for` loops.